### ISSUE TITLE
Add official RP4 addresses to deployment CI as env

### ISSUE THREAD

**Issue by rouzwelt** [rainlanguage/rain.orderbook#901]
sushi lib uses official RP contract addresses internally, some routes may include them depending on the route, and since our CI deploy a fresh RP contract everytime, at times we might get some unexpected reverts if such routes are given to the bot to process.

solution: add official RP4 address as env to the CI/ gh secrets, here is the list of RP4 contract per chain:
https://github.com/rainlanguage/sushiswap/blob/34fbe7d482e43cdb847b8a5116d0d2b6e18e6d18/packages/sushi/src/config/route-processor.ts#L304

**Comment by hardyjosh** [comment]
<img width="553" alt="image" src="https://github.com/user-attachments/assets/3ae5d5e1-7024-4624-a9bd-7bc985339f45">

**Comment by thedavidmeister** [comment]
yeah @rouzwelt i'm not sure if i got an answer to that question in the screenshot @hardyjosh posted

if the sushi router is hardcoded to the singular RP4 contract, why don't we see it break already, because we've been deploying and using our own copy of RP4 for a long time now

**Comment by rouzwelt** [comment]
> yeah @rouzwelt i'm not sure if i got an answer to that question in the screenshot @hardyjosh posted
> 
> if the sushi router is hardcoded to the singular RP4 contract, why don't we see it break already, because we've been deploying and using our own copy of RP4 for a long time now

@thedavidmeister we discussed this on chat, the reason is that, not all produced routes contain a piece to the RP4 contract address, some more complex routes do, but not all, mostly the simple ones containing 1 or 2 pool

**Comment by thedavidmeister** [comment]
🤖 ai:producer

Re-verified against main 2026-07-29: still present — and the one hook that could have carried an official address was removed.

- `script/Deploy.sol:91-104` still deploys raindex's **own** RP4 from vendored Sushi creation code (`src/lib/deploy/LibRouteProcessor4CreationCode.sol`).
- `src/generated/RouteProcessor4.pointers.sol:17` pins `DEPLOYED_ADDRESS = 0x6E2d0e71d900474b262E545Bc4C98b71ab368d21` — one Zoltu address for all chains, not any of Sushi's per-chain official RP4 addresses. `src/concrete/arb/RouteProcessorRaindexV6ArbOrderTaker.sol:31` hardcodes it.
- No official-RP4 env/secret in `.github/workflows` (`manual-sol-artifacts.yaml` passes only `suite`), and no FFI script fetching Sushi's `route-processor.ts` (solution 2 also unimplemented).
- Merged commit `3d1ca8f3c` (2026-03-14, "Convert RouteProcessor4 to Zoltu deterministic deployment") states: *"Remove `DEPLOY_ROUTE_PROCESSOR_4_ADDRESS` env var — arb suite now uses the deterministic address from generated pointers."*

On the open question above ("why don't we see it break already"): rain.solver builds route calldata against Sushi's **official** RP4 (`src/state/chain.ts:80`, sourced from `sushi/config`, passed as `RPAddr` at `src/core/modes/raindex/index.ts:126`) while the destination arb contract executes through raindex's own RP4 — the address mismatch is live across the two repos.

One factual softening vs the 2024 text: deployment is now CREATE2-idempotent, so it is no longer "a fresh RP contract every time" — but the address still is not Sushi's.
